### PR TITLE
Readme Sections on Demo

### PR DESCRIPTION
**Pull request by thinkocapo**
can preview the readme here
https://github.com/thinkocapo/android/blob/auth/README.md

We now have the following sections in README with useful knowledge and Gotcha's
- ANR
- SESSIONS
- Release Health Testing
- How to Make New Release
- How to Upgrade SDK

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,44 +86,86 @@ In this case, you can fix that by updating Sentry's server. To do that:
...
     minSdkVersion 21
     targetSdkVersion 29
-    versionCode 11
-    versionName "1.2"
+    versionCode 13
+    versionName "1.3"
+}
 ```
-This would make for a release of `1.1.0 (11) com.example.vu@androidh1.2+11`. 
+This would make for a release of `1.3.0 (13) com.example.vu@androidh1.3+13`.
```
Comment (marandaneto) [inline README.md:114]: `com.example.vu.android@1.3+13`

File: README.md [hunk 2]
```diff
@@ -85,44 +86,86 @@ In this case, you can fix that by updating Sentry's server. To do that:
...
 The version code is unique. This is already part of build system in Android. The app won't compile without it.
-The info in AndroidManifest.xml will override what's in build.gradle.
 
-**Other**
+Optional - Setting the release here is good if you really had a reason to override, eg. Paid vs Free versions of your apps
+This is not generally for a customer. It's for testing so I can quickly iterate new releases while I'm testing. This info in AndroidManifest.xml will override what's in build.gradle.
```
Comment (marandaneto) [inline README.md:118]: well, it is also a customer issue, but probably not used that much.
why would they use that?
eg
1: the pattern package@name+version is new from Sentry, so they could use this feature to keep the previous way as Sentry forces this one now.
2: they would like to match versions for Android and iOS, so they can force a release name
3: as mentioned above, a release for free and paid version
etc...

File: README.md [hunk 3]
```diff
@@ -85,44 +86,86 @@ In this case, you can fix that by updating Sentry's server. To do that:
...
+```
+<meta-data android:name="io.sentry.release" android:value="io.sentry.sample@1.0.0+1" />
+```
+
+## How To Upgrade SDK
+1. increment sdk number in src/build.gradle like `implementation 'io.sentry:sentry-android:2.1.4'`
```
Comment (marandaneto) [inline README.md:124]: 2.1.6 is out

File: README.md [hunk 4]
```diff
@@ -85,44 +86,86 @@ In this case, you can fix that by updating Sentry's server. To do that:
...
+4. `make all` will do a new `./gradlew build`
+
+## ANR
 Sometimes you'll see extra ANR events, because you have setting set to 3 seconds
-Hard to compare Total Number of Crashes to a report in Discover on handled:no and the release, because when a crash happens, you have to wait for the device to come back online again. 
-There are some other technical reasons as well, which are still being sorted out. 
-For instance, if you're evering filtering, sampling or Rate Limiting events/crashes out, then it's possible that the Sessions data isn ot getting filtered/sampled and so your Crash Free rate will appear higher than it actually is.
+Hard to compare Total Number of Crashes to a report in Discover on handled:no and the release, because when a crash happens, you have to wait for the device to come back online again.
+There are some other technical reasons as well, which are still being sorted out.
+For instance, if you're ever filtering, sampling or Rate Limiting events/crashes out, then it's possible that the Sessions data isn ot getting filtered/sampled and so your Crash Free rate will appear higher than it actually is.
```
Comment (marandaneto) [inline README.md:133]: yep, sessions are never sampled, just to confirm.

Comment (thinkocapo) [reply README.md:133]: thanks

**Comment by thinkocapo** [comment]
@marandaneto thanks I made these updates directly on 'master in the README just now.

**Pull request merged**